### PR TITLE
cyberglove_calibrator updated to work on kinetic. Needed dependencies…

### PR DESCRIPTION
**Pull request by azompas**
… were added.

### CODE REVIEW

File: advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(sr_gui_cyberglove_calibrator)
-find_package(catkin REQUIRED COMPONENTS rospy rqt_gui rqt_gui_py std_msgs std_srvs diagnostic_msgs sr_robot_msgs sr_visualization_icons)
+find_package(catkin REQUIRED COMPONENTS rospy rqt_gui rqt_gui_py std_msgs std_srvs diagnostic_msgs sr_robot_msgs sr_visualization_icons python-scipy)
```
Comment (guihomework) [inline advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt:3]: @azompas  why is that needed ? I did a similar fix (before seeing yours) and did not have to add this dependency (but had to add dependency to cyberglove to get cyberglove.srv)

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py [hunk 2]
```diff
@@ -320,32 +323,29 @@ def __init__(self, context):
...
+        btn_calibrate = QtWidgets.QPushButton()
         btn_calibrate.setText("Calibrate")
         btn_calibrate.setToolTip("Calibrate the current selected step")
         btn_calibrate.setIcon(
             QtGui.QIcon(rootPath + '/images/icons/calibrate.png'))
         btn_layout.addWidget(btn_calibrate)
-        btn_frame.connect(btn_calibrate, QtCore.SIGNAL(
-            'clicked()'), self.calibrate_current_step)
-        self.btn_save = QtGui.QPushButton()
+        btn_frame.btn_calibrate.clicked.connect(self.calibrate_current_step)
```
Comment (guihomework) [inline advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:335]: btn_frame gets the layout containing the btn_calibrated added later, so I wonder if this can work (similar for btn_save and btn_load)

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py [hunk 3]
```diff
@@ -366,29 +366,28 @@ def calibrate_current_step(self):
...
 
         self.calibrer.write_calibration_file(filename)
 
         # QMessageBox returns 0 for yes
-        if QtGui.QMessageBox.information(self._widget,
-                                         "Load new Calibration",
-                                         "Do you want to load the new calibration file?",
-                                         "yes",
-                                         "no") == 0:
+        if QtWidgets.QMessageBox.information(self._widget,
```
Comment (guihomework) [inline advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:377]: I had issues there, 
```
TypeError: QMessageBox.information(QWidget, str, str, QMessageBox.StandardButtons buttons=QMessageBox.Ok, QMessageBox.StandardButton defaultButton=QMessageBox.NoButton): argument 4 has unexpected type 'str'
```
so had to fix that differently than here

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py [hunk 4]
```diff
@@ -366,29 +366,28 @@ def calibrate_current_step(self):
...
+            filename = QtWidgets.QFileDialog.getOpenFileName(
                 self._widget, 'Open Calibration', '')
             if "" == filename:
                 return
 
         self.calibrer.load_calib(str(filename))
 
-        self.emit(QtCore.SIGNAL("messageToStatusbar(QString)"),
-                  "New Cyberglove Calibration Loaded.")
+        self.messageToStatusbar['QString'].emit("New Cyberglove Calibration Loaded.")
```
Comment (guihomework) [inline advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:393]: does not work
```
self.messageToStatusbar['QString'].emit("New Cyberglove Calibration Loaded.")
AttributeError: 'SrGuiCybergloveCalibrator' object has no attribute 'messageToStatusbar'
```

Comment (guihomework) [reply advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:393]: Actually this does not work for sometime already  https://github.com/ros-visualization/rqt_robot_plugins/issues/92#issuecomment-134722338
So this code was either not tested for long or did not reach this point due to other errors

File: advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt (hunk 1 above)
Comment (azompas) [reply advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt:3]: I did this to install the python-scipy in case that it is not already installed in the system

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py (hunk 3 above)
Comment (azompas) [reply advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:377]: I couldn't find anything that specifies that there should be any other difference in this. Your solution might be applicable here but I did not have this error

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py (hunk 4 above)
Comment (azompas) [reply advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:393]: probably this is bag that still exists. Actually when I was testing it, I launched the glove with no error but I was getting an error from rqt that there is no glove.

File: advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt (hunk 1 above)
Comment (guihomework) [reply advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt:3]: Indeed, well spotted, without this system install, we get
ImportError: No module named scipy.optimize, I did not look good enough when trying to find the usage.

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py (hunk 3 above)
Comment (guihomework) [reply advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:377]: There is more to do, I don't have a no button on my solution, because I think an information does not have two buttons by default, it should be a question messagebox isn't it ?

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py (hunk 4 above)
Comment (guihomework) [reply advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:393]: Pressing the button was actually locking up at the wait_for_service on my side (without glove), and did not come to the message_to_statusbar in a first place.

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py [hunk 5]
```diff
@@ -90,36 +94,35 @@ def set_description(self, description):
...
         self.step_describer = StepDescriber(self)
 
-        self.list = QtGui.QListWidget()
+        self.list = QtWidgets.QListWidget()
         first_item = self.refresh_list()
-        self.connect(self.list, QtCore.SIGNAL(
-            'itemClicked(QListWidgetItem*)'), self.step_choosed)
-        self.list.setViewMode(QtGui.QListView.ListMode)
-        self.list.setResizeMode(QtGui.QListView.Adjust)
+        self.list.itemClicked['QListWidgetItem*'].connect(self.step_choosed)
```
Comment (guihomework) [inline advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:123]: I get this error with your code 
```
AttributeError: 'QListWidget' object has no attribute 'setItemSelected'
```
and indeed one should refer to the item now, my solution fixes it

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py (hunk 2 above)
Comment (guihomework) [reply advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:335]: confirmed, by getting this error
```
btn_frame.btn_calibrate.clicked.connect(self.calibrate_current_step)
AttributeError: 'QFrame' object has no attribute 'btn_calibrate'
```

File: advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py (hunk 5 above)
Comment (azompas) [reply advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:123]: You get this message when you start it on rqt or when you press a specific button?
I remember I could start it on rqt and press all buttons but it would do nothing because it said there is no glove

Comment (guihomework) [reply advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py:123]: Sorry, the error is actually in a part you did not change, but affected by qt5 changes as well
https://github.com/shadow-robot/sr-visualization/blob/4dbf89f49bd2c070c2113c7194cccba433a62bd1/advanced/sr_gui_cyberglove_calibrator/src/sr_gui_cyberglove_calibrator/cyberglove_calibrator.py#L129

this method does not exist anymore see https://doc.qt.io/qt-5/qlistwidget.html

I don't know why you did not get the error, but this is not valid pyqt5 so should be fixed.

File: advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt (hunk 1 above)
Comment (mikramarc) [inline advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt:3]: So this python-scipy is not actually needed here because it's only a run dependency, right? (as we did for the hand calibrator before)

**Review by mikramarc — CHANGES_REQUESTED** [review]

File: advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt (hunk 1 above)
Comment (guihomework) [reply advanced/sr_gui_cyberglove_calibrator/CMakeLists.txt:3]: I think this whole PR was agreed to not be merged but rather use mine including the fixes and some more https://github.com/shadow-robot/sr-visualization/pull/156
